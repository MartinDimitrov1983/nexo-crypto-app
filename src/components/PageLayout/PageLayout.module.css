.layout {
    display: flex;
    flex-direction: column;
    align-items: center;
    min-height: 100vh;
    background-color: #f5f6f8;
    padding: 0 100px;
}

.title {
    margin: 30px 0;
    color: #5b6064;
}
