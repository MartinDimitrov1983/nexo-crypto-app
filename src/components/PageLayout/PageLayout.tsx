import React, { ReactNode } from 'react';
import { Loading } from '../';
import styles from './PageLayout.module.css';

interface PageLayoutProps {
    title?: string;
    children: ReactNode;
    loading: Boolean;
}

const PageLayout = ({ title, loading, children }: PageLayoutProps) => {
    return (
        <div className={styles.layout}>
            {loading && <Loading />}
            <h1 className={styles.title}>{title}</h1>

            {children}
        </div>
    );
};

export default PageLayout;
