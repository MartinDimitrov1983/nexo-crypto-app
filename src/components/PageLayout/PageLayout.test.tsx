import React from 'react';
import { render } from '@testing-library/react';

import PageLayout from './PageLayout';

test('Snapshot PageLayout', () => {
    const { asFragment } = render(
        <PageLayout title="test" children={null} loading={false} />,
    );
    expect(asFragment()).toMatchSnapshot();
});
