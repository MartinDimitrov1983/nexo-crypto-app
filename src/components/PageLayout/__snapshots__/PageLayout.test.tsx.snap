// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot PageLayout 1`] = `
<DocumentFragment>
  <div
    class="layout"
  >
    <h1
      class="title"
    >
      test
    </h1>
  </div>
</DocumentFragment>
`;
